### ISSUE TITLE
Grid with details should notify the users of the Expand/Collapse functionality

### ISSUE THREAD

**Issue by Alex-Bubblemaster** [telerik/kendo-ui-core#4480]
### Enhancement request

Ticket with ID 1340493 

### Reproduction of the problem

When using a screen reader the expand/collapse options of the master rows are not announced to the user.

[Official hierarchy demo](https://demos.telerik.com/kendo-ui/grid/hierarchy)

### Current behavior

The hierarchy cells are excluded from navigation line 102 from kendo.grid.js, if they are not, the readers are able to read [the labels](https://github.com/telerik/kendo-ui-core/issues/3369)
 
` DATA_CELL = ":not(.k-group-cell):not(.k-hierarchy-cell):visible"`

### Expected/desired behavior

The readers should read the expand/collapse option as it does for the [grouped grid](https://demos.telerik.com/kendo-ui/grid/keyboard-navigation)

### Environment

* **Kendo UI version:** 2018.2.620

**Comment by Alex-Bubblemaster** [comment]
#3369

**Comment by ag-petrov** [comment]
With the new functionality the user will be able to navigate to the expand/collapse cell which holds a button. This button has an aria-label indicating the possible action(expand or collapse). Thus the reades will notify the user for action he may want to perform.
